server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -0,0 +1,260 @@
+package com.epam.catgenome.manager.reference;
+
+import com.epam.catgenome.entity.reference.Chromosome;
+import com.epam.catgenome.entity.reference.StrandedSequence;
+import com.epam.catgenome.entity.reference.motif.Motif;
+import com.epam.catgenome.entity.reference.motif.MotifSearchRequest;
+import com.epam.catgenome.entity.reference.motif.MotifSearchResult;
+import com.epam.catgenome.entity.reference.motif.MotifSearchType;
+import com.epam.catgenome.entity.track.Track;
+import com.epam.catgenome.manager.gene.parser.StrandSerializable;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static com.epam.catgenome.component.MessageHelper.getMessage;
+
+@Service
+@Slf4j
+public class MotifSearchManager {
+
+    @Autowired
+    private ReferenceGenomeManager referenceGenomeManager;
+
+    private static final int TRACK_LENGTH = 100;
+
+    public Track<StrandedSequence> fillTrackWithMotifSearch(final Track<StrandedSequence> track,
+                                                            final String motif,
+                                                            final StrandSerializable strand) {
+        final List<StrandedSequence> result = search(
+                MotifSearchRequest.builder()
+                        .startPosition(track.getStartIndex())
+                        .endPosition(track.getEndIndex())
+                        .chromosomeId(track.getChromosome().getId())
+                        .motif(motif)
+                        .searchType(MotifSearchType.REGION)
+                        .pageSize(0)
```

Comment:
better to use Integer.MAX_VALUE here instead of 0, to show that we will load all result that we have